### PR TITLE
[BEAM-2288] Refine DSL interface

### PR DESCRIPTION
**Pull request by XuMingmin**
Refine DSL interface as design doc of BEAM-2010:
1. rename BeamSqlRunner to BeamSQLEnvironment;
2. Move pipeline as external parameter, to integrate with none-DSL components;

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by XuMingmin** [comment]
R: @xumingming 
CC: @davorbonaci @jbonofre @takidau

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/BeamSQLEnvironment.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+sqlEnv.registerPCollectionAsTable(String tableName, phase1Stream);
+
+//apply more queries, even based on phase1Stream
+
+pipeline.run().waitUntilFinish();
+ * }
+ * </pre>
+ *
+ * <h1>2. BeamSQL as CLI</h1>
+ * This feature is on planning, and not ready yet.
```
Comment (xumingming) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/BeamSQLEnvironment.java]: By CLI you mean a command client like `mysql`, you input a `SQL query`, it gives you a tabular result?

Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/BeamSQLEnvironment.java]: Yes, `CLI` means a command client, the result behavior is TBD, I expect result returned for bounded, and run on detach mode for unbounded.

**Comment by xumingming** [comment]
LGTM

**Comment by jbonofre** [comment]
It has been merged on `DSL_SQL` branch. Can you please close this pull request ? Thanks !

**Comment by XuMingmin** [comment]
Thanks @jbonofre  @xumingming !

**Pull request closed without merging**